### PR TITLE
[Tutorials] Replacing the old "Your First NFT" tutorial entry function calls to use aptos_token.move and new indexer queries

### PR DESCRIPTION
**Pull request by xbtmatt**
### Description

The old "your first NFT" tutorial hadn't been updated in a long time and it used very old calls and the `0x3` token contract. I've kept the structure the same but used the new `0x4::aptos_token.move` contract along with new indexer queries.

Added documentation tags to the typescript and python sdks for referenced functions.

Built off of #9422 since it uses the `transfer_token` function in the tutorial.

### Test Plan

Run the tutorials and look at the output (it's very simple)

### CODE REVIEW

File: ecosystem/python/sdk/aptos_sdk/aptos_token_client.py [hunk 1]
```diff
@@ -515,6 +516,23 @@ async def mint_soul_bound_token(
...
+            [TypeTag(StructTag.from_str("0x4::token::Token"))],
+            [
+                TransactionArgument(token, Serializer.struct),
+                TransactionArgument(to, Serializer.struct)
+            ],
+        )
+
+        signed_transaction = await self.client.create_bcs_signed_transaction(
+            sender, TransactionPayload(payload)
+        )
```
Comment (gregnazario) [inline ecosystem/python/sdk/aptos_sdk/aptos_token_client.py:533]: https://github.com/aptos-labs/aptos-core/pull/9422#pullrequestreview-1563692972

File: ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    "Alice's simple collection",
+    collectionName,
+    "https://alice.com",
+    maxSupply,
+    {
+      royaltyNumerator: 5,
+      royaltyDenominator: 100,
+    },
+  ); // <:!:section_4
+  await provider.aptosClient.waitForTransaction(txnHash1, { checkSuccess: true });
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts:65]:
```suggestion
  await provider.waitForTransaction(txnHash1, { checkSuccess: true });
```

File: ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    alice,
+    collectionName,
+    "Alice's simple token",
+    tokenName,
+    "https://aptos.dev/img/nyan.jpeg",
+    [],
+    [],
+    [],
+  ); // <:!:section_5
+  await provider.aptosClient.waitForTransaction(txnHash2, { checkSuccess: true });
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts:79]:
```suggestion
  await provider.waitForTransaction(txnHash2, { checkSuccess: true });
```

File: ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+  // Print the collection data.
+  // :!:>section_6
+  const collectionData = (await provider.getCollectionData(alice.address(), collectionName)).current_collections_v2[0];
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts:83]: getCollectionData() is an indexer query, later in the code we check whether indexer and devnet in sync - if not we skip tests. so if they are not synced, this test would "fail". maybe best would be to check indexer and devnet chain id and then tun this example if they are synced

Comment (0xmaayan) [reply ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts:83]: that's still a thing, it is rare and with the local env should be completely gone

**Review by 0xmaayan — APPROVED** [review]
TS looks good to me

**Review by davidiw — APPROVED** [review]

File: ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts (hunk 4 above)
Comment (xbtmatt) [reply ecosystem/typescript/sdk/examples/typescript/simple_aptos_token.ts:88]: ok gotcha, I fixed it

**Pull request merged**